### PR TITLE
[dy] Test source and destination connection

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Test source and destination connection from integration pipeline view. I added a `test_connection` method to the source and destination classes, so all subclasses need to implement this method.

Screenshot 2022-11-01 at 9 40 52 PM

Screenshot 2022-11-02 at 3 11 47 PM

# Tests

tested locally with all sources and destinations

cc: @wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_preparation/models/pipelines/integration_pipeline.py [hunk 1]
```diff
@@ -108,6 +108,29 @@ def transformer_file_path(self) -> str:
...
+            if file_path:
+                run_args = [
+                    PYTHON,
+                    file_path,
+                    '--config_json',
+                    simplejson.dumps(config),
+                    '--test_connection',
+                ]
+
+                proc = subprocess.run(run_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=5)
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/models/pipelines/integration_pipeline.py:128]: nit: break the line if it's exceeds 100 characters

File: mage_ai/data_preparation/models/pipelines/integration_pipeline.py [hunk 2]
```diff
@@ -108,6 +108,29 @@ def transformer_file_path(self) -> str:
...
+                proc.check_returncode()
+        except subprocess.CalledProcessError as e:
+            message = e.stderr.decode("utf-8")
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/models/pipelines/integration_pipeline.py:131]: nit: single quote

File: mage_integrations/mage_integrations/sources/amplitude/__init__.py [hunk 3]
```diff
@@ -63,6 +66,11 @@ def get_table_key_properties(self, stream_id):
     def get_valid_replication_keys(self, stream_id):
         return VALID_REPLICATION_KEYS[stream_id]
 
+    def test_connection(self):
+        today = datetime.utcnow()
+        self.connection.load(start_date=today, end_date=today)
+
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/amplitude/__init__.py:72]: nit: remove extra blank line

**Review by wangxiaoyou1993 — COMMENTED** [review]
How do we show errors when connection test fails?

**Comment by dy46** [comment]
@wangxiaoyou1993 I added the error message

File: mage_ai/server/server.py [hunk 4]
```diff
@@ -602,9 +602,9 @@ def start_server(
 
     if manage:
         os.environ[MANAGE_ENV_VAR] = '1'
-    else:
-        # Start a subprocess for scheduler
-        scheduler_manager.start_scheduler()
+    # else:
```
Comment (wangxiaoyou1993) [inline mage_ai/server/server.py:605]: uncomment this?

File: mage_integrations/mage_integrations/sources/base.py [hunk 5]
```diff
@@ -185,13 +191,12 @@ def process(self) -> None:
                 catalog = self.catalog or self.discover(streams=self.selected_streams)
                 self.sync(catalog)
         except Exception as err:
-            message = f'{self.__class__.__name__} process failed with error {err}.'
-            self.logger.exception(message, tags=dict(
+            # message = f'{self.__class__.__name__} process failed with error {sys.exc_info()}.'
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/base.py:194]: uncomment this?

File: mage_integrations/mage_integrations/sources/base.py [hunk 6]
```diff
@@ -185,13 +191,12 @@ def process(self) -> None:
...
+            self.logger.exception('error', tags=dict(
                 error=str(err),
                 errors=traceback.format_stack(),
                 message=traceback.format_exc(),
             ))
-            raise Exception(message)
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/base.py:194]: why don't we raising exception here?

File: mage_integrations/mage_integrations/sources/stripe/__init__.py [hunk 7]
```diff
@@ -1179,7 +1181,9 @@ def new_request(self, method, url, params=None, headers=None):
     # set the config and state in prior to check the authentication in the discovery mode itself.
     configure_stripe_client()
 
-    if source.discover_mode:
+    if source.should_test_connection:
+        pass
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/stripe/__init__.py:1185]: I saw you implemented the test_connection method. why don't you call it here?

File: mage_integrations/mage_integrations/sources/titanic/__init__.py [hunk 8]
```diff
@@ -21,6 +21,13 @@ def load_data(
 
         yield rows
 
+    def test_connection(self):
+        url = 'https://raw.githubusercontent.com/mage-ai/datasets/master/titanic_survival.csv'
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/titanic/__init__.py:25]: make this url a constant

File: mage_integrations/mage_integrations/sources/stripe/__init__.py (hunk 7 above)
Comment (dy46) [reply mage_integrations/mage_integrations/sources/stripe/__init__.py:1185]: When the main method for the stripe source is called, it tries to connect to the stripe client by default, so calling it again would be redundant.

**Comment by dy46** [comment]
@wangxiaoyou1993 addressed your comments

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**